include/dlpack/dlpack.h

```diff
@@ -67,7 +67,7 @@ typedef struct {
  */
 typedef struct {
   /*! \brief Type code of base types */
-  uint8_t code;
+  DLDataTypeCode code;
```

Comment:
So, at a high level, I have some doubts about the complexity of the DLDataType. For example, what if someone marks DLDataTypeCode as kFloat but then sets the bits to, say, 17?

I actually like the current design of TensorFlow and Caffe2 - in the sense that we explicitly list out the data types instead of using bits. The reason is that there are not much possible choices of the bits value, so it is easier to define FLOAT64, FLOAT32, FLOAT16 instead of doing error-prone (kFloat, 23). 

By the way, I still like the existence of DLDataType because of lanes - I think we should keep it, but kill the bits field. Thoughts?